app/services/populate_database.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+class PopulateDatabase
+  def initialize(register)
+    @registers_client ||= RegistersClient::RegisterClientManager.new(cache_duration: 60)
+    @register = register
+  end
+
+  def bulk_remove_existing_records(register, entry_type, record_keys)
+    unless record_keys.empty?
+      Record.where(spina_register_id: register.id, entry_type: entry_type, key: record_keys).delete_all
+    end
+  end
+
+  def bulk_save(entries, records)
+    Entry.import!(entries)
+    Record.import!(records)
+  end
+
+  def populate_register
+    Delayed::Worker.logger.info("Updating #{@register.name} in database")
+    register_data = @registers_client.get_register(@register.name.parameterize, @register.register_phase.downcase)
```

Comment:
You're going to have a problem here, in that getting each register is going to download the entire contents of the register every time the job is invoked. This may be unavoidable, but it's because a new `RegisterClientManager` is being constructed each time the job is run, so every time it believes it hasn't seen any data before and gets the whole register again. If we could store the `RegisterClientManager` as we were before in a class variable and have the same instance 
 accessible each time the job is run, we can avoid this.